src/Xamarin.Android.Build.Tasks/Tasks/Javac.cs

```diff
@@ -60,12 +60,25 @@ protected override string GenerateCommandLineCommands ()
 			cmd.AppendSwitchIfNotNull ("-J-Dfile.encoding=", "UTF8");
 
 			cmd.AppendFileNameIfNotNull (string.Format ("@{0}", TemporarySourceListFile));
-			cmd.AppendSwitchIfNotNull ("-target ", JavacTargetVersion);
-			cmd.AppendSwitchIfNotNull ("-source ", JavacSourceVersion);
+
+			if (int.TryParse (JavacSourceVersion, out int sourceVersion) &&
+					int.TryParse (JavacTargetVersion, out int targetVersion) &&
+					JavacSupportsRelease ()) {
+				cmd.AppendSwitchIfNotNull ("--release ", Math.Max (sourceVersion, targetVersion).ToString ());
+			} else {
+				cmd.AppendSwitchIfNotNull ("-target ", JavacTargetVersion);
+				cmd.AppendSwitchIfNotNull ("-source ", JavacSourceVersion);
+			}
 
 			return cmd.ToString ();
 		}
 
+		bool JavacSupportsRelease ()
+		{
+			var jdkInfo = new JdkInfo (Path.Combine (ToolPath, ".."));
```

Comment:
Will this do file i/o? Probably can be a bit slow on Windows?
* https://github.com/dotnet/android-tools/blob/main/src/Xamarin.Android.Tools.AndroidSdk/JdkInfo.cs

Could we use the `$(_JdkVersion)` property instead?

https://github.com/dotnet/android/blob/6754e0524437edb231f8a98c399ff2a870ca3747/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Tooling.targets#L32

I think it would be set prior to this task.